lib/core_extensions.rb

```diff
@@ -51,6 +59,19 @@ def before_destroy(record)
         false
       end
     end
+
+    def check_permissions(record, klass, klass_name, what)
+      association = record.class.reflect_on_association(klass.to_sym).klass
+      association = Host if association == Host::Managed
+      permission = "view_#{klass_name}"
```

Comment:
That could be fixed by either splitting it into two methods (one that returns nil, another that wraps and raises an exception) or adding a flag?